### PR TITLE
complete components, text, and village

### PR DESCRIPTION
**Pull request by eristhea**
again, straightforward.
refactor: `TooltipContext` -> `TooltipType`
I found a class in `Item` that much better matches the context name, and so this needed a new name. I was considering both `TooltipProperties` and `TooltipType`, and chose `Type`, though it could go either way imo.

### CODE REVIEW

File: mappings/net/minecraft/client/item/TooltipType.mapping
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/client/item/TooltipType.mapping:1]: How about `TooltipConfig`?

File: mappings/net/minecraft/component/type/DyedColorComponent.mapping
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/component/type/DyedColorComponent.mapping:1]: I think all `showInTooltip` should be `shownInTooltip` in this class, `UnbreakableComponent`, and `AttributeModifiersComponent` (this is how it is in `ItemEnchantmentsComponent`).

see https://github.com/QuiltMC/quilt-mappings/pull/559#discussion_r1525123993

**Review by supersaiyansubtlety — COMMENTED** [review]
A couple suggestions

File: mappings/net/minecraft/client/item/TooltipType.mapping
Comment (eristhea) [reply mappings/net/minecraft/client/item/TooltipType.mapping:1]: that works with me!

**Comment by eristhea** [comment]
both suggestions accepted!

**Review by supersaiyansubtlety — APPROVED** [review]

**Review by OroArmor — APPROVED** [review]

**Pull request merged**